private/buf/cmd/buf/command/mod/modprune/modprune.go

```diff
@@ -28,61 +28,39 @@ import (
 	"github.com/bufbuild/buf/private/pkg/rpc"
 	"github.com/bufbuild/buf/private/pkg/storage/storageos"
 	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
-)
-
-const (
-	dirFlagName = "dir"
 )
 
 // NewCommand returns a new prune Command.
 func NewCommand(
 	name string,
 	builder appflag.Builder,
 ) *appcmd.Command {
-	flags := newFlags()
 	return &appcmd.Command{
-		Use:   name,
+		Use:   name + " <directory>",
 		Short: "Prunes unused dependencies from the " + buflock.ExternalConfigFilePath + " file.",
-		Args:  cobra.NoArgs,
+		Long:  `The first argument is the directory of the local module to prune. If no argument is specified, defaults to "."`,
+		Args:  cobra.MaximumNArgs(1),
 		Run: builder.NewRunFunc(
 			func(ctx context.Context, container appflag.Container) error {
-				return run(ctx, container, flags)
+				return run(ctx, container)
 			},
 			bufcli.NewErrorInterceptor(),
 		),
-		BindFlags: flags.Bind,
 	}
 }
 
-type flags struct {
-	// for testing only
-	Dir string
-}
-
-func newFlags() *flags {
-	return &flags{}
-}
-
-func (f *flags) Bind(flagSet *pflag.FlagSet) {
-	flagSet.StringVar(
-		&f.Dir,
-		dirFlagName,
-		".",
-		"The directory to operate in. For testing only.",
-	)
-	_ = flagSet.MarkHidden(dirFlagName)
-}
-
 // run tidy to trim the buf.lock file for a specific module.
 func run(
 	ctx context.Context,
 	container appflag.Container,
-	flags *flags,
 ) error {
+	directoryInput, err := bufcli.GetInputValue(container, "", "", "", ".")
```

Comment:
We actually needed to have `flags.InputHashtag` and then pass it here, basically cobra ignores anything after `-#`. While we only take directories, we probably want to future-proof this. See `push.go` for an example.